tests/test_freqtradebot.py

```diff
@@ -3545,32 +3545,29 @@ def test_get_real_amount_no_trade(default_conf, buy_order_fee, caplog, mocker, f
                    caplog)
 
 
-def test_get_real_amount_stake(default_conf, trades_for_order, buy_order_fee, fee, mocker):
-    trades_for_order[0]['fee']['currency'] = 'ETH'
-
-    mocker.patch('freqtrade.exchange.Exchange.get_trades_for_order', return_value=trades_for_order)
-    amount = sum(x['amount'] for x in trades_for_order)
-    trade = Trade(
-        pair='LTC/ETH',
-        amount=amount,
-        exchange='binance',
-        fee_open=fee.return_value,
-        fee_close=fee.return_value,
-        open_rate=0.245441,
-        open_order_id="123456"
-    )
-    freqtrade = get_patched_freqtradebot(mocker, default_conf)
-
-    # Amount does not change
-    assert freqtrade.get_real_amount(trade, buy_order_fee) == amount
-
-
-def test_get_real_amount_no_currency_in_fee(default_conf, trades_for_order, buy_order_fee,
-                                            fee, mocker):
+@pytest.mark.parametrize(
+    'fee_cost, fee_currency, fee_reduction_amount, use_ticker_rate, expected_log', [
+        (None, 'ETH', 0, True, None),
+        (0.004, None, 0, True, None),
+        (0.00094518, "BNB", 0, True, (
+            'Fee for Trade Trade(id=None, pair=LTC/ETH, amount=8.00000000, open_rate=0.24544100,'
+            ' open_since=closed) [buy]: 0.00094518 BNB - rate: None'
+        )),
+        (0.004, "LTC", 0.004, False, (
+            'Applying fee on amount for Trade(id=None, pair=LTC/ETH, amount=8.00000000, '
+            'open_rate=0.24544100, open_since=closed) (from 8.0 to 7.996).'
+        )),
+        (0.008, None, 0, True, None),
```

Comment:
Now, this is essentially duplicate of https://github.com/freqtrade/freqtrade/pull/5601/files#diff-f0d1952492d82eb12612502e7426fa75194b47a35bf0a04a9ad1af0f924ee25bR3551

Mentioned more here:
https://github.com/freqtrade/freqtrade/pull/5601/files#r714345676

Reply:
i don't understand what your trying to say here

Reply:
This test case is the same as ` (0.004, None, 0, True, None),` This is no longer testing what it was originally testing. An option to keep it the same would be to pass in the fee object instead, `{"cost": 0.008}` and `{"cost": 0.004, "currency": None}` for the previous other test case to capture testing the same thing as before

Reply:
I THINK that's the better way to test this, yes - it's also clearer to understand - IMHO

Reply:
> This test case is the same as ` (0.004, None, 0, True, None),` This is no longer testing what it was originally testing. An option to keep it the same would be to pass in the fee object instead, `{"cost": 0.008}` and `{"cost": 0.004, "currency": None}` for the previous other test case to capture testing the same thing as before

Sounds good, make it test what it was originally testing